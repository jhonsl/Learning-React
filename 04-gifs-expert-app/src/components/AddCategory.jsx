import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setinputValue] = useState('');
    const onInputChange = ({ target }) => setinputValue(target.value);
    const onSubmit = (event) => {
        event.preventDefault();
        inputValue.trim() != '' && onNewCategory( inputValue );
        setinputValue('');
    };

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input 
                type="text" 
                placeholder="Search Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}