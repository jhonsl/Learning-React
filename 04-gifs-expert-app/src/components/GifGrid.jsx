import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            { isLoading && <h2>Cargando...</h2> }

            <div className="card-grid">
                {
                    images.map(({ id, ...rest }) => (
                        <GifItem 
                            key={ id } 
                            { ...rest }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}