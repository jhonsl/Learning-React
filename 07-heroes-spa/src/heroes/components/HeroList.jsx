import React, { useMemo } from 'react'
import { getHeroesByPublihser } from '../helpers/getHeroesByPublisher'
import { HeroCard } from './HeroCard';

export const HeroList = ({ publisher }) => {
    const heroes = useMemo(() => getHeroesByPublihser(publisher), [publisher]);

    return (
        <div className='row rows-cols-1 row-cols-md-3 g-3'>
            {
                heroes.map(heroe => 
                    <HeroCard key={heroe.id} {...heroe}/>
                )
            }
        </div>
    )
}
