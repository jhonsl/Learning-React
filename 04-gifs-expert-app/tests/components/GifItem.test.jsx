import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('probando GifItem.jsx', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Probando el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url }/>);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostar la imagen con el URÃ‘ y el ALT indicado', () => {
        render(<GifItem title={ title } url={ url }/>);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('debe de mostar el titulo en el componente', () => {
        render(<GifItem title={ title } url={ url }/>);
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});