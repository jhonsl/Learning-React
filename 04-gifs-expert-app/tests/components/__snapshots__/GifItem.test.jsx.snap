// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando GifItem.jsx Probando el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
